Assets/Scripts/UI/InventoryUI.cs

```diff
@@ -1,10 +1,13 @@
-﻿using Inventory;
+using System;
 using System.Collections.Generic;
+using System.Linq;
+using Interfaces;
+using Inventory;
 using UnityEngine;
 
 namespace UI
 {
-    public class InventoryUI : MonoBehaviour
+    public class InventoryUI : MonoBehaviour, IItemHandler
```

Comment:
I think `IItemHandler` should be implemented by the `Inventory` class itself, and not by `InventoryUI` 🤔 Afaik the inventory class also already defines item update events, so you could either build on that foundation or rewrite it so that it fits the interface :)